### PR TITLE
cnosdb 1.0.2 (new formula)

### PR DESCRIPTION
**Pull request by xmh1011**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/cnosdb.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  desc "Open source distributed time series database"
+  homepage "https://www.cnosdb.com"
+  url "https://github.com/cnosdb/cnosdb/archive/refs/tags/v1.0.2.tar.gz"
+  sha256 "ae412d0944b64c9b39dc1edc66f7b6f712b85bc5afad354c12b135ae71017100"
+  license "MIT"
+  head "https://github.com/cnosdb/cnosdb.git", branch: "main"
+  depends_on "go" => :build
+  def install
+    ENV["GOBIN"] = buildpath
+    system "go", "install", "./..."
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:11]: This should use `go build` like all other formulae
```suggestion
    ENV["GOBIN"] = buildpath
    system "go", "install", "./..."
```

File: Formula/cnosdb.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          <string>#{var}/log/cnosdb.log</string>
+          <key>SoftResourceLimits</key>
+          <dict>
+            <key>NumberOfFiles</key>
+            <integer>10240</integer>
+          </dict>
+        </dict>
+      </plist>
+    EOS
+  end
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:56]: This should not be a caveat.

File: Formula/cnosdb.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  test do
+    (testpath/"config.toml").write shell_output("#{bin}/cnosdb config")
+    inreplace testpath/"config.toml" do |s|
+      s.gsub! %r{/.*/.cnosdb/data}, "#{testpath}/cnosdb/data"
+      s.gsub! %r{/.*/.cnosdb/meta}, "#{testpath}/cnosdb/meta"
+      s.gsub! %r{/.*/.cnosdb/wal}, "#{testpath}/cnosdb/wal"
+    end
+    begin
+      pid = fork do
+        exec "#{bin}/cnosdb --config #{testpath}/config.toml"
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:66]:
```suggestion
        exec bin/"cnosdb", "--config", testpath/"config.toml"
```

File: Formula/cnosdb.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      end
+      sleep 60
+      output = shell_output("curl -Is localhost:8086/ping")
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:69]: this should be using `free_port` to get a free port to use.

**Review by SMillerDev — CHANGES_REQUESTED** [review]

File: Formula/cnosdb.rb (hunk 2 above)
Comment (xmh1011) [reply Formula/cnosdb.rb:56]: What should I use? I don't know, can you tell me? Thank you.

Comment (SMillerDev) [reply Formula/cnosdb.rb]: User instructions should be in the upstream documentation and the plist should be replaced by a `service do` block

Comment (xmh1011) [reply Formula/cnosdb.rb]: Thank you,sir.

File: Formula/cnosdb.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      -X main.version=#{version}
+      -X main.commit=#{Utils.git_short_head(length: 10)}
+      -X main.date=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(output: bin/"cnosdb", ldflags: ldflags), "./cmd/cnosdb"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-cli", ldflags: ldflags), "./cmd/cnosdb-cli"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-inspect", ldflags: ldflags), "./cmd/cnosdb-inspect"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-tools", ldflags: ldflags), "./cmd/cnosdb-tools"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-ctl", ldflags: ldflags), "./cmd/cnosdb-ctl"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-meta", ldflags: ldflags), "./cmd/cnosdb-meta"
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:28]:
```suggestion
    %w[cnosdb cnosdb-cli cnosdb-inspect cnosdb-tools cnosdb-ctl cnosdb-meta].each do |cmd|
      system "go", "build", *std_go_args(output: bin/cmd, ldflags: ldflags), "./cmd/#{cmd}"
    end
```

File: Formula/cnosdb.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      s.gsub! %r{/.*/.cnosdb/meta}, "#{testpath}/cnosdb/meta"
+      s.gsub! %r{/.*/.cnosdb/wal}, "#{testpath}/cnosdb/wal"
+    end
+    begin
+      pid = fork do
+        exec bin/"cnosdb", "--config", testpath/"config.toml"
+      end
+      sleep 60
+      output = shell_output("curl -Is localhost:#{cnosdb_port}/ping")
+      assert_match output
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:44]:
```suggestion
      output = shell_output("curl -Is localhost:#{cnosdb_port}/ping")
      assert_match "something", output
```

File: Formula/cnosdb.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    system "go", "build", *std_go_args(output: bin/"cnosdb-cli", ldflags: ldflags), "./cmd/cnosdb-cli"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-inspect", ldflags: ldflags), "./cmd/cnosdb-inspect"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-tools", ldflags: ldflags), "./cmd/cnosdb-tools"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-ctl", ldflags: ldflags), "./cmd/cnosdb-ctl"
+    system "go", "build", *std_go_args(output: bin/"cnosdb-meta", ldflags: ldflags), "./cmd/cnosdb-meta"
+  end
+  test do
+    cnosdb_port = free_port
+    (testpath/"config.toml").write shell_output("#{bin}/cnosdb config")
+    inreplace testpath/"config.toml" do |s|
```
Comment (SMillerDev) [inline Formula/cnosdb.rb:33]: I think this is missing the port setting

Comment (xmh1011) [reply Formula/cnosdb.rb]: yes, I will fix it. Thank you.

File: Formula/cnosdb.rb (hunk 6 above)
Comment (xmh1011) [reply Formula/cnosdb.rb]: Thank you, I have fixed it.

**Pull request closed without merging**